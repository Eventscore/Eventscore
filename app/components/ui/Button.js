//<Button text={'Server is down'} />
import React, { Component, PropTypes } from 'react';
import { Button } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts, AppSizes } from '@theme/';

class CustomButton extends Component {
  static propTypes = {
    small: PropTypes.bool,
    large: PropTypes.bool,
    outlined: PropTypes.bool,
    backgroundColor: PropTypes.string,
    onPress: PropTypes.func.isRequired,
    icon: PropTypes.shape({
      name: PropTypes.string,
    }),
  }

  static defaultProps = {
    small: false,
    large: false,
    outlined: false,
    icon: {},
    backgroundColor: null,
  }

  buttonProps = () => {
    // Defaults
    const props = {
      title: 'Coming Soon...',
      color: '#fff',
      fontWeight: 'bold',
      onPress: this.props.onPress,
      fontFamily: AppFonts.base.family,
      fontSize: AppFonts.base.size,
      borderRadius: AppSizes.borderRadius,
      raised: true,
      buttonStyle: {
        padding: 12,
        marginLeft: 0,
        marginRight: 0,
      },
      ...this.props,
      backgroundColor: this.props.backgroundColor || AppColors.brand.primary,
      small: false,
      large: false,
      icon: (this.props.icon && this.props.icon.name)
        ? {
          size: 14,
          ...this.props.icon,
        } : null,
    };

    // Overrides
    // Size
    if (this.props.small) {
      props.fontSize = 12;
      props.buttonStyle.padding = 8;

      if (props.icon && props.icon.name) {
        props.icon = {
          size: 14,
          ...props.icon,
        };
      }
    }
    if (this.props.large) {
      props.fontSize = 20;
      props.buttonStyle.padding = 15;

      if (props.icon && props.icon.name) {
        props.icon = {
          size: 20,
          ...props.icon,
        };
      }
    }

    // Outlined
    if (this.props.outlined) {
      props.raised = false;
      props.backgroundColor = this.props.backgroundColor || 'transparent';
      props.color = AppColors.brand.primary;
      props.buttonStyle.borderWidth = 1;
      props.buttonStyle.borderColor = AppColors.brand.primary;

      if (props.icon && props.icon.name) {
        props.icon = {
          color: AppColors.brand.primary,
          ...props.icon,
        };
      }
    }

    return props;
  }

  render = () => <Button {...this.buttonProps()} />;
}

export default CustomButton;
