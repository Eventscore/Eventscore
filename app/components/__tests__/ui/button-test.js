/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '@ui/Button';

// Default Button Press function
const onPress = () => Alert.alert('Hey Tester');

it('Button renders correctly', () => {
  const tree = renderer.create(
    <Button />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Button with onPress renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Button with Text renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} title={'Test Button'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Button with Icon renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} icon={{ name: 'code' }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Outlined Button renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} outlined />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Outlined Button with Icon renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} outlined icon={{ name: 'code' }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Small Button renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} small />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Small Button with Icon renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} small icon={{ name: 'code' }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Large Button renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} small />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Large Button with Icon renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={onPress} large icon={{ name: 'code' }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
