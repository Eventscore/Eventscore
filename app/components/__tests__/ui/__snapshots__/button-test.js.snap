// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Button with Icon renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¡¯
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Button with Text renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`Button with onPress renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Large Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 8,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Large Button with Icon renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¡¯
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Outlined Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "borderRadius": 2,
        },
        false,
        Object {
          "padding": 12,
        },
        Object {
          "borderColor": "#0E4EF8",
          "borderWidth": 1,
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#0E4EF8",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Outlined Button with Icon renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "borderRadius": 2,
        },
        false,
        Object {
          "padding": 12,
        },
        Object {
          "borderColor": "#0E4EF8",
          "borderWidth": 1,
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0E4EF8",
            "fontSize": 14,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¡¯
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#0E4EF8",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Small Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 8,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;

exports[`Small Button with Icon renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 8,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¡¯
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }
    >
      Coming Soon...
    </Text>
  </View>
</View>
`;
